pkg/cache/tas_flavor.go

```diff
@@ -77,29 +82,54 @@ func (c *TASFlavorCache) snapshot(ctx context.Context) *TASFlavorSnapshot {
 	requiredLabelKeys = append(requiredLabelKeys, c.Levels...)
 	err := c.client.List(ctx, nodeList, requiredLabels, requiredLabelKeys)
 	if err != nil {
-		log.Error(err, "failed to list nodes for TAS", "nodeLabels", c.NodeLabels)
+		return nil, fmt.Errorf("failed to list nodes for TAS: %w", err)
 	}
-	return c.snapshotForNodes(log, nodeList.Items)
+	r, err := labels.NewRequirement(kueuealpha.TASLabel, selection.DoesNotExist, nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to build requirement for non-TAS pods: %w", err)
+	}
+	podListOpts := &client.ListOptions{}
+	podListOpts.LabelSelector = labels.NewSelector()
+	podListOpts.LabelSelector = podListOpts.LabelSelector.Add(*r)
+	podList := corev1.PodList{}
```

Comment:
```suggestion
	pods := corev1.PodList{}
```

nits

Reply:
ok, but in that case I also rename nodeList to nodes

Reply:
Yeah, we should do it as well